seed/seed.js

```diff
@@ -5,63 +5,169 @@
 const childProcess = require('child_process');
 const path = require('path');
 
+const { getAllCourses } = require('../resources/grabs');
 const models = require('../server/db/index.js');
 const { db } = models;
 const userList = require('./seedlings/userSeed');
-const groupList = require('./seedlings/groupSeed');
+const { groupList } = require('./seedlings/groupSeed');
 const eventList = require('./seedlings/eventSeed');
 const postList = require('./seedlings/postSeed');
+const zipList = require('./seedlings/zipSeed');
+
+const findModel = async (name) => {
+  const topic = await models.Topic.findOne({
+    where: {
+      name
+    }
+  })
+  return topic;
+}
+
+const hasKeys = (obj) => {
+  let count = 0
+  for (let key in obj) {
+    if (key) count++
+  }
+  if (count >= 1) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+const topicNameIds = async (modelArr) => {
+  try {
+    const topicAssociation = {};
+    modelArr.forEach(async t => {
+      try {
+        const top = await t
+        if (!top) {
+          return
+        } else {
+          topicAssociation[top.name] = top.id
+        }
+      } catch (err) {
+        console.error(err);
+      }
+    });
+    return topicAssociation;
+  } catch (err) {
+    console.error(err);
+  }
+}
 
 // This spins everything up, all the functions are below
 doTheSeeding();
 
 async function doTheSeeding() {
   try {
     await db.sync({ force: true });
+    await getAllCourses();
+
+    const testPrep = findModel('Test Prep');
+    const math = findModel('Math');
+    const artsAndHumanities = findModel('Arts and humanities');
+    const eAndF = findModel('Economics and finance');
+    const science = findModel('Science');
+    const computing = findModel('Computing');
+
+    const topicArr = [testPrep, math, artsAndHumanities, eAndF, science, computing];
+    const topics = await topicNameIds(topicArr);
     const users = await Promise.all(genList(userList, models.User));
-    const groups = await Promise.all(genList(groupList, models.Group));
     const events = await Promise.all(genList(eventList, models.Event));
-    postList.forEach(p => {
-      p.userId = users[Math.floor(Math.random() * users.length)].id;
-      p.groupId = groups[Math.floor(Math.random() * groups.length)].id;
-    });
 
-    await Promise.all(genList(postList, models.Post));
+    const topicNames = {}
+    if (hasKeys(topics)) {
```

Comment:
why not just directly iterate over topics and use hasOwnProperty? This is duplicating work.